nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/TestRepository.java

```diff
@@ -444,6 +449,68 @@ public void itCanHandleComplexPropertiesWithJson() throws IOException {
         assertEquals("file", document.getTitle());
     }
 
+    /**
+     * Dates can be handled properly if and only if the date is given as {@link java.lang.String}.
+     */
+    @Test
+    public void itCanHandleDateAsString() {
+        GregorianCalendar calendar = new GregorianCalendar(2017, Calendar.MAY, 4, 3, 2, 1);
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSX");
```

Comment:
I use `.SS` instead of `.SSS` because I exported a `document.xml` from Nuxeo Server 8.3 from an event (schema type `vevent`) and here's the excerpt:

```xml
  <schema xmlns:vevent="http://www.nuxeo.org/ecm/schemas/AgendaEvent/" name="vevent">
    <vevent:location><![CDATA[My office]]></vevent:location>
    <vevent:transp><![CDATA[OPAQUE]]></vevent:transp>
    <vevent:dtstart><![CDATA[2017-01-24T11:00:00.00Z]]></vevent:dtstart>
    <vevent:categories><![CDATA[]]></vevent:categories>
    <vevent:dtend><![CDATA[2017-01-25T11:00:00.00Z]]></vevent:dtend>
    <vevent:status><![CDATA[CONFIRMED]]></vevent:status>
  </schema>
  <schema xmlns:dc="http://www.nuxeo.org/ecm/schemas/dublincore/" name="dublincore">
    <dc:description><![CDATA[This is an important meeting]]></dc:description>
    <dc:creator><![CDATA[Administrator]]></dc:creator>
    <dc:modified><![CDATA[2017-01-23T16:04:49.00Z]]></dc:modified>
    <dc:lastContributor><![CDATA[Administrator]]></dc:lastContributor>
    <dc:created><![CDATA[2017-01-23T16:04:49.00Z]]></dc:created>
    <dc:title><![CDATA[Meeting]]></dc:title>
    <dc:subjects/>
    <dc:contributors>
      <item><![CDATA[Administrator]]></item>
    </dc:contributors>
  </schema>
```

As you can see, only 2 digits are kept in the date part millisecond. This is also proved by the following assertions.

OK for `XXX`, I'll update it.